backend/compact-connect/lambdas/staff-users/tests/function/test_data_model/test_client.py

```diff
@@ -0,0 +1,442 @@
+        )
+
+    def test_update_user_permissions_board_to_compact_admin(self):
+        # The sample user looks like board staff in aslp/oh
+        user_id = UUID(self._load_user_data())
+
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        resp = client.update_user_permissions(
+            compact='aslp',
+            user_id=user_id,
+            compact_action_additions={'admin'},
+            jurisdiction_action_removals={
+                'oh': {'write'}
+            }
+        )
+
+        self.assertEqual(user_id, resp['userId'])
+        self.assertEqual(
+            {
+                'actions': {'read', 'admin'},
+                'jurisdictions': {}
+            },
+            resp['permissions']
+        )
+        # Checking that we're getting the whole object, not just changes
+        self.assertFalse(
+            {'type', 'userId', 'compact', 'attributes', 'permissions', 'dateOfUpdate'} - resp.keys()
+        )
+
+    def test_update_user_permissions_compact_to_board_admin(self):
+        from boto3.dynamodb.types import TypeDeserializer
+
+        with open('tests/resources/dynamo/user.json', 'r') as f:
+            user_data = TypeDeserializer().deserialize({'M': json.load(f)})
+
+        user_id = UUID(user_data['userId'])
+        # Convert our canned user into a compact admin
+        user_data['permissions'] = {
+            'actions': {'read', 'admin'},
+            'jurisdictions': {}
+        }
+        self._table.put_item(
+            Item=user_data
+        )
+
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        resp = client.update_user_permissions(
+            compact='aslp',
+            user_id=user_id,
+            compact_action_removals={'admin'},
+            jurisdiction_action_additions={
+                'oh': {'write', 'admin'}
+            }
+        )
+
+        self.assertEqual(user_id, resp['userId'])
+        self.assertEqual(
+            {
+                'actions': {'read'},
+                'jurisdictions': {
+                    'oh': {'write', 'admin'}
+                }
+            },
+            resp['permissions']
+        )
+        # Checking that we're getting the whole object, not just changes
+        self.assertFalse(
+            {'type', 'userId', 'compact', 'attributes', 'permissions', 'dateOfUpdate'} - resp.keys()
+        )
+
+    def test_update_user_attributes(self):
+        # The sample user looks like board staff in aslp/oh
+        user_id = UUID(self._load_user_data())
+
+        from data_model.client import UserClient
+
+        client = UserClient(self.config)
+
+        resp = client.update_user_attributes(
+            user_id=user_id,
+            attributes={
+                'givenName': 'Bob',
+                'familyName': 'Smith'
+            }
+        )
+        self.assertEqual(1, len(resp))
+        user = resp[0]
```

Comment:
why would it ever return more than one item (the user)?

Reply:
A user has a record in the DB for each compact they have permissions for. In this case, the user is only in `aslp`, so they should only get one record back.